packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -308,6 +309,23 @@ export interface AwsCustomResourceProps {
    * ID for the function's name. For more information, see Name Type.
    */
   readonly functionName?: string;
+
+  /**
+   * The vpc to provision the lambda function in.
+   *
+   * @default - the function is not provisioned inside a vpc.
+   */
+  readonly vpc?: ec2.IVpc;
+
+  /**
+   * Which subnets from the VPC to place the lambda function in.
+   *
+   * Only used if 'vpc' is supplied. Note: internet access for Lambdas
+   * requires a NAT gateway, so picking Public subnets is not allowed.
+   *
+   * @default - the Vpc default strategy if not specified
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
```

Comment:
What happens if this is supplied when VPC is not? Does it fail silently/fail at deployment/fail at synth?

Reply:
I think it's just ignored? Which seems to be a pattern (here, with `lambda.Function`, and `customresources.Provider`). It's not especially ideal but it seems to be semi-predictable. https://github.com/aws/aws-cdk/blob/17f1ec881ba8fb300bd4cf8674a87640ab05c31a/packages/%40aws-cdk/aws-lambda/lib/function.ts#L1129 is the only use of `props.vpcSubnets` I see in the `Function` construct, so no validation seems to be performed there to guard against it being passed senselessly.

If something is done to validate this prop here, it'd probably also need to work for the `Provider` construct. https://github.com/aws/aws-cdk/blob/17f1ec881ba8fb300bd4cf8674a87640ab05c31a/packages/%40aws-cdk/custom-resources/lib/provider-framework/provider.ts#L205-L219

Reply:
Having a value that just gets ignored is not a pattern I think we should be perpetuating. It's unfortunate that we haven't been validating this already because adding it to the right place (in lambda) could potentially break someone's code. We should, however, have enough test coverage added so that we know the answer to if/how this will fail.

Reply:
I agree. I've opened #21369 to at least have the conversation about throwing an Error. And maybe at least it could fall back to an `Annotations.of(this).addWarning()` either here or there. But I'll save the specific of that discussion for that PR.